docusaurus/docs/create-a-plugin/extend-a-plugin/add-resource-handler.md

```diff
@@ -0,0 +1,184 @@
+
+An alternative to using the [`CallResource` method shown in the above example](#add-resource-handler-for-data-source-plugins) is to use [`httpadapter`](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/resource/httpadapter) and [`ServeMux`](https://pkg.go.dev/net/http#ServeMux) as shown below:
+
+```
+package mydatasource
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/resource/httpadapter"
+)
+
+type MyDatasource struct {
+	resourceHandler backend.CallResourceHandler
+}
+
+func New() *MyDatasource {
+	ds := &MyDatasource{}
+	mux := http.NewServeMux()
+	mux.HandleFunc("/namespaces", ds.handleNamespaces)
+	mux.HandleFunc("/projects", ds.handleProjects)
+	ds.resourceHandler := httpadapter.New(mux)
+	return ds
+}
+
+func (d *MyDatasource) CallResource(ctx context.Context, req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
+	return d.resourceHandler.CallResource(ctx, req)
+}
+
+func (d *MyDatasource) handleNamespaces(rw http.ResponseWriter, req *http.Request) {
+	_, err := rw.Write([]byte(`{ namespaces: ["ns-1", "ns-2"] }`))
+	if err != nil {
+		return
+	}
+	rw.WriteHeader(http.StatusOK)
+}
+
+func (d *MyDatasource) handleProjects(rw http.ResponseWriter, req *http.Request) {
+	_, err := rw.Write([]byte(`{ projects: ["project-1", "project-2"] }`))
+	if err != nil {
+		return
+	}
+	rw.WriteHeader(http.StatusOK)
+}
+```
+
+:::note
+
+Using some other HTTP router library with above example should be straightforward. Just replace the use of [`ServeMux`](https://pkg.go.dev/net/http#ServeMux) with another router.
+
+:::
+
+### Additional examples
+
+Some other examples of using the [`httpadapter`](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/resource/httpadapter) package can be found for some of the built-in Grafana data sources:
+
+- Test data source [grafana/testdata.go](https://github.com/grafana/grafana/blob/30aa24a18396f61bfbcbfbeee78b1c0d23685fc3/pkg/tsdb/testdatasource/testdata.go#L41-L42) at 30aa24a18396f61bfbcbfbeee78b1c0d23685fc3 - See code at the [grafana/grafana](https://github.com/grafana/grafana/blob/30aa24a18396f61bfbcbfbeee78b1c0d23685fc3/pkg/tsdb/testdatasource/testdata.go#L41-L42) repo.
+- Test data source [grafana/resource_handler.go](https://github.com/grafana/grafana/blob/30aa24a18396f61bfbcbfbeee78b1c0d23685fc3/pkg/tsdb/testdatasource/resource_handler.go#L18) at 30aa24a18396f61bfbcbfbeee78b1c0d23685fc3 - See code at the [grafana/grafana](https://github.com/grafana/grafana/blob/30aa24a18396f61bfbcbfbeee78b1c0d23685fc3/pkg/tsdb/testdatasource/resource_handler.go#L18) repo.
+
+### What if you need access to the backend plugin context?
```

Comment:
I would suggest this section to go before the examples section.

Reply:
done.